src/test/java/uk/gov/hmcts/reform/blobrouter/services/storage/BulkScanSasTokenCacheTest.java

```diff
@@ -0,0 +1,107 @@
+package uk.gov.hmcts.reform.blobrouter.services.storage;
+
+import com.azure.storage.common.Utility;
+import com.azure.storage.common.implementation.Constants;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.hmcts.reform.blobrouter.clients.bulkscanprocessor.BulkScanProcessorClient;
+import uk.gov.hmcts.reform.blobrouter.clients.bulkscanprocessor.SasTokenResponse;
+
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.BDDMockito.given;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(MockitoExtension.class)
+class BulkScanSasTokenCacheTest {
+
+    @Mock
+    private BulkScanProcessorClient bulkScanProcessorClient;
+
+    private BulkScanSasTokenCache bulkScanContainerClientCache;
+
+    private long refreshSasBeforeExpiry = 30;
+
+    @BeforeEach
+    private void setUp() {
+        this.bulkScanContainerClientCache = new BulkScanSasTokenCache(
+            bulkScanProcessorClient,
+            refreshSasBeforeExpiry
+        );
+    }
+
+    @Test
+    void should_create_sas_token() {
```

Comment:
Suggesion: add the `...when` part to this and next method, to explain what conditions we're checking the behaviour for.

Reply:
added